### PR TITLE
Utilities: Implement a netstat command

### PR DESCRIPTION
**Pull request by brapru**
First time contributing to Serenity, welcome all the critiques!

### CODE REVIEW

**Comment by bcoles** [comment]
`out()` and `outln()` are preferred over `printf("...")` and `printf("...\n")`.

Presumably the output will be unsorted. `/proc/net/*` data is not sorted and I don't see any sorting in this code. You can use `AK/QuickSort.h` for this. There are a few examples in the code base. Here's a rudimentary psuedo-code implementation:

```cpp
    Vector<JsonValue> sorted_regions = json.value().as_array().values();
    quick_sort(sorted_regions, [](auto& a, auto& b) {
        return a.as_object().get("local_port").to_u32() < b.as_object().get("local_port").to_u32();
    });
```

**Comment by brapru** [comment]
Thanks for the feedback. Updated with suggested changes

File: Userland/Utilities/netstat.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    protocol_column = add_column("Proto", Alignment::Left, 5);
+    bytes_in_column = add_column("Bytes-In", Alignment::Right, 9);
+    bytes_out_column = add_column("Bytes-Out", Alignment::Right, 9);
+    local_address_column = add_column("Local Address", Alignment::Left, 20);
+    peer_address_column = add_column("Peer Address", Alignment::Left, 20);
+    state_column = add_column("State", Alignment::Left, 9);
+
+    auto print_column = [](auto& column, auto& string) {
+        if (!column.width) {
+            out("{}", string.characters());
```
Comment (awesomekling) [inline Userland/Utilities/netstat.cpp:82]: No need for `.characters()`:
```cpp
out("{}", string);
```

File: Userland/Utilities/netstat.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            return;
+        }
+        if (column.alignment == Alignment::Right) {
+            auto line = String::formatted(StringView { "{:<{1}}  " }, string.characters(), column.width);
+            out(line);
```
Comment (awesomekling) [inline Userland/Utilities/netstat.cpp:87]: No need to create a temporary string:
```cpp
out("{:<{1}}  "sv, string, column.width);
```

File: Userland/Utilities/netstat.cpp [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        } else {
+            auto line = String::formatted(StringView { "{:<{1}}  " }, string.characters(), column.width);
+            out(line);
```
Comment (awesomekling) [inline Userland/Utilities/netstat.cpp:90]: Same as above.

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Userland/Utilities/netstat.cpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            out("{:<{1}}  "sv, string, column.width);
+        }
+    };
+
+    if (!has_protocol_flag || flag_tcp || flag_udp) {
+        out("Active Internet connections ");
+
+        if (flag_all)
+            outln("(servers and established)");
+        else {
```
Comment (linusg) [inline Userland/Utilities/netstat.cpp:97]:
```suggestion
        if (flag_all) {
            outln("(servers and established)");
        } else {
```

Mixing non-braced if and braced else looks weird.

File: Userland/Utilities/netstat.cpp [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            auto bytes_out = if_object.get("bytes_out").to_string();
+            auto peer_address = if_object.get("peer_address").to_string();
+            auto peer_port = if_object.get("peer_port").to_string();
+            auto formatted_peer_address = String::formatted("{}:{}", peer_address, peer_port);
+            auto local_address = if_object.get("local_address").to_string();
+            auto local_port = if_object.get("local_port").to_string();
+            auto formatted_local_address = String::formatted("{}:{}", local_address, local_port);
+            auto state = if_object.get("state").to_string();
+
+            if (!flag_all && (((state == "Listen") && !flag_list) || (!(state == "Listen") && flag_list)))
```
Comment (linusg) [inline Userland/Utilities/netstat.cpp:138]:
```suggestion
            if (!flag_all && ((state == "Listen" && !flag_list) || (state != "Listen" && flag_list)))
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Utilities/netstat.cpp (hunk 4 above)
Comment (brapru) [reply Userland/Utilities/netstat.cpp]: Fixed 👍

File: Userland/Utilities/netstat.cpp (hunk 5 above)
Comment (brapru) [reply Userland/Utilities/netstat.cpp]: Thanks for the feedback! Made suggested changes 👍

**Pull request merged**